src/coverlet.tracker/CoverageTracker.cs

```diff
@@ -42,16 +56,30 @@ public static void MarkExecuted(string file, string evt)
         [ExcludeFromCodeCoverage]
         public static void CurrentDomain_ProcessExit(object sender, EventArgs e)
         {
+            Dictionary<string, bool> encounteredFile = new Dictionary<string, bool>();
+
             lock (_events)
             {
-                foreach (var files in _events)
+                foreach (var localEvents in _events)
                 {
-                    using (var fs = new FileStream(files.Key, FileMode.Create))
-                    using (var sw = new StreamWriter(fs))
+                    lock (localEvents)
                     {
-                        foreach (var evt in files.Value)
+                        foreach (var files in localEvents)
                         {
-                            sw.WriteLine($"{evt.Key},{evt.Value}");
+                            FileMode mode = FileMode.Open;
```

Comment:
Another thought: will `FileMode.Open` position the file for writing at the end? I can't quite make it out from the documenation at https://docs.microsoft.com/en-us/dotnet/api/system.io.filemode?view=netcore-2.1

If not, this will open the file for writing at the start and overwrite what was written the last time.  It could be better to use `FileMode.Append` instead (see same page above) to ensure that data is always added to the file.

An alternative is to keep the open file streams in `encounteredFile`, but at the price of not eing able to use `using` and the streams must be disposed manually at the end.

Reply:
I think you are right here. I'll open the file with the AppendMode.